### ISSUE TITLE
datasets

### ISSUE THREAD

**Issue by wwwhdxghdgxjh** [Kalaimani02/ANN_in_Verilog#2]
Hello, the author. Is the function of this code to classify MNIST datasets? How can MNIST datasets be converted into Verilog usable data

**Comment by Kalaimani02** [comment]
I have included the Test Dataset folder right. It has the pixel intensities of each image in binary format in a text file (which is Verilog usable data). I have also included one more folder ImgToBinaryConverter. It has the python code to convert the image to the required binary format. The MNIST dataset has images of size 28x28, giving 784-pixel values in binary format for one image.
Also as you said the function of the designed neural network is to classify the MNIST datasets.

**Comment by wwwhdxghdgxjh** [comment]
Why can't I implement the classification of MNIST datasets? I have run through your code, and the results are as follows. Can you help me answer the problem
![1654690001(1)](https://user-images.githubusercontent.com/105864752/172612461-cbd72fbf-e04c-42bc-a2a5-e99b758602c0.png)

**Comment by wwwhdxghdgxjh** [comment]
Weight_ Is there an error in this code when memory reads the weight file, so the weight seems to be unreadabl
![1654695626(1)](https://user-images.githubusercontent.com/105864752/172631645-33bcc2e4-4ddb-4e10-b123-adf7edef4725.png)
e

**Comment by Kalaimani02** [comment]
Did you changed the no of layers or no of neurons in each layer ?

**Comment by wwwhdxghdgxjh** [comment]
No, I just ran your source code. Is it because of the test file? I used your test file and did not import the complete MNIST dataset

**Comment by wwwhdxghdgxjh** [comment]
The python file you wrote is to convert a picture of MNIST datasets, instead of converting all MNIST datasets into a TXT document

**Comment by Kalaimani02** [comment]
> No, I just ran your source code. Is it because of the test file? I used your test file and did not import the complete MNIST dataset

No its not because of the the test. My test files should work good. You need not to import complete MNIST dataset

**Comment by Kalaimani02** [comment]
If you have followed the same procedure as mentioned in the video I attached before it should work

**Comment by wwwhdxghdgxjh** [comment]
Is a four layer fully connected neural network built

**Comment by Kalaimani02** [comment]
Yes. It's a 4 layer fully connected Neural Network

**Comment by wwwhdxghdgxjh** [comment]
Author, thank you for answering my questions. I'd like to ask another question about your code: when the data is transferred to the neurons of each layer, the results are accumulated after the weights and inputs are multiplied, and then the accumulated results are transferred to the neurons of the next layer. I don't see the operation process of accumulation in your code

**Comment by Kalaimani02** [comment]
Checkout the 'sum' variable in neuron.v file

**Comment by wwwhdxghdgxjh** [comment]
Author, did you write the test file wron
![1655347260(1)](https://user-images.githubusercontent.com/105864752/173979828-268d73b8-cc47-4456-a073-d2349953f69b.png)
g

**Comment by wwwhdxghdgxjh** [comment]
$readmemb reads the file name and the filename is the pointer. Is there an error here

**Comment by Kalaimani02** [comment]
No filename is a vector which has the test data file to be read. It automatically loops through the 30 test data files. Just go below the test bench code. You will find it

**Comment by wwwhdxghdgxjh** [comment]
I can't read the data. I put the data in the Modelsim folder and didn't change your test file. Why is thi
![1655714932(1)](https://user-images.githubusercontent.com/105864752/174564026-df732d70-4ca0-457d-a22b-2bdb47ad94ec.png)
s

**Comment by wwwhdxghdgxjh** [comment]
> 作者，你写的测试文件wron ![1655347260(1)](https://user-images.githubusercontent.com/105864752/173979828-268d73b8-cc47-4456-a073-d2349953f69b.png) g

The two parameters in $readmemb are pointers. You cannot traverse all test data

**Comment by Kalaimani02** [comment]
Hi. Go through the test bench file completely. If you understand that then your issues will be clarified
